torch/autograd/function.py

```diff
@@ -54,6 +55,65 @@ def save_for_backward(self, *tensors: torch.Tensor):
         """
         self.to_save = tensors
 
+    def save_for_forward(self, *tensors: torch.Tensor):
+        r"""Saves given tensors for a future call to :func:`~Function.forward`.
+
+        This should be only called from inside the :func:`forward` method.
+        This method should only be called with tensors. Unlike `save_for_backward`,
+        the passed tensors need not be an input or output.
```

Comment:
Interesting, I also updated the save_for_backward description. I know we never enforced it, but is the reason that we had the note originally (before the saved variable updates) because there was overhead from to making a shallow copy?

Reply:
The core idea for this is to avoid memory leak. And so this was not needed to avoid this one.
Indeed it is unnecessary overhead when you pass in non-input/output Tensors.